doc-site/pages/docs/ui/index.mdx

```diff
@@ -33,25 +30,158 @@ for [antd](https://ant.design) or you can implement your own.
+doesn't have to be.
+
+```js
+class Person extends viewModelFactory(
+    {
+        id: new Field(),
+        isActive: new BooleanField({ label: 'Active?' }),
+    },
+    { pkFieldName: 'id' }
+) {}
+
+function PersonDetail({ person }) {
+    return (
+        <dl>
+            <dt>{Person.fields.isActive.label}</dt>
+            <dd>
+                <FieldFormatter field={person._f.isActive} />
+            </dd>
+        </dl>
+    );
+}
+```
+
+In the above example [BooleanFormatter](doc:BooleanFormatter) will be used as that's the default formatter for [BooleanField](doc:BooleanField).
+The advantage of this over hard-coding the formatter everywhere is you can change how a particular field type is formatted
+in a central location. If you wanted to change how `BooleanField` is formatted you could implement your own formatter. All
+a formatter is is a component that accepts, at minimum, a `value` prop:
 
 ```js
-function UpperCaseFormatter({ value }) {
-    if (!value) {
-        return value;
+function BooleanEmojiFormatter({ value }) {
+    if (value == null || value == '') {
+        return <>'❓'</>;
     }
-    return value.toUpperCase();
+    return <>{value ? '✅' : '❌'}</>;
 }
 ```
 
-When you use [FieldFormatter](doc:FieldFormatter) it just works out what formatter component to use based on what the
-`getFormatterForField` function on your [UiProvider](doc:UiProvider) returns.
+To use this formatter everywhere a [BooleanField](doc:BooleanField) appears provide you own implementation of
+`getFormatterForField`:
+
+```js
+import { getFormatterForField as defaultGetFormatterForField } from '@prestojs/ui';
+
+function getFormatterForField(field) {
+    if (field instanceof BooleanField) {
+        // You can just return the formatter component directly but it's good practice to return
+        // the component & `field.getFormatterProps` which allows per field customisations.
+        return [BooleanEmojiFormatter, field.getFormatterProps()];
+    }
+    // Fall back to the default implementation
+    return defaultGetFormatterForField(field);
+}
+
+// Then pass this to the `UiProvider` that wraps your app
+<UiProvider getFormatterForField={getFormatterForField}>{...}</UiProvider>
+```
+
+As shown in the above example `getFormatterForField` can return a 2-element array of the form `[component, defaultProps]`.
+This allows you to inject props used on a component based on the field. We could rewrite the above to use the options
+provided by the existing [BooleanFormatter](doc:BooleanFormatter) component:
+
+```js
+import { getFormatterForField as defaultGetFormatterForField } from '@prestojs/ui';
+
+function getFormatterForField(field) {
+    if (field instanceof BooleanField) {
+        return [
+            BooleanFormatter,
+            { trueLabel: '✅', falseLabel: '❌', blankLabel: '❓', ...field.getFormatterProps() },
```

Comment:
Would it make more sense to destructure first, then override properties?

Reply:
No - this allows a specific field to override the general defaults